quickstart/docker/image/add-router.sh

```diff
@@ -0,0 +1,182 @@
+fi
+
+if [[ "${ZITI_EDGE_CONTROLLER_PORT}" != "" ]]; then
+   ctrl_port=${ZITI_EDGE_CONTROLLER_PORT}
+else
+   ctrl_port="1280"
+fi
+
+if [[ "${ZITI_EDGE_CONTROLLER_MANAGEMENT_PORT}" != "" ]]; then
+   mgmt_port=${ZITI_EDGE_CONTROLLER_MANAGEMENT_PORT}
+else
+   mgmt_port="6262"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_LAN_PORT}" != "" ]]; then
+   lan_port=${ZITI_EDGE_ROUTER_LAN_PORT}
+else
+   lan_port="ens3"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_PORT}" != "" ]]; then
+   rport=${ZITI_EDGE_ROUTER_PORT}
+else
+   rport="443"
+fi
+
+if [[ "${ZITI_EDGE_ROUTER_NAME}" != "" ]]; then
+   er_name=${ZITI_EDGE_ROUTER_NAME}
+else
+   er_name="none"
+fi
+
+router_ip=$(ip addr show ${lan_port} | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)
+
+echo Use ${router_ip} as internal IP
+
+cat <<EOT >config.yml
+v: 3
+
+identity:
+   cert: "certs/identity.cert.pem"
+   server_cert: "certs/internal.chain.cert.pem"
+   key: "certs/internal.key.pem"
+   ca: "certs/intermediate-chain.pem"
+
+edge:
+  csr:
+    country: US
+    locality: Charlotte
+    organization: Netfoundry
+    organizationalUnit: ADV-DEV
+    #province: NC
+    sans:
+      dns:
+        - "localhost"
+      ip:
+        - "127.0.0.1"
+        - "router_ip"
+        - "external_router_ip"
+
+ctrl:
+  endpoint: tls:edgecontroller:ctrl_port
+
+link:
+  dialers:
+    - binding: transport
+
+listeners:
+  - binding: edge
+    address: tls:0.0.0.0:rport
+    options:
+      advertise: advertise_ip:rport
+      maxQueuedConnects:      50
+      maxOutstandingConnects: 100
+      connectTimeoutMs:       3000
+  - binding: tunnel
+    options:
+      svcPollRate: 15s
+      resolver: udp://router_ip:53
+      dnsSvcIpRange: 100.64.0.1/10
+      lanIf: lan_port
+EOT
+
+#create path for router binaries, certs and congig
+mkdir -p certs
+
+#update config template with external ip of new router
+if [[ "${ZITI_EDGE_ROUTER_IP_OVERRIDE}" != "" ]]; then
+   echo ${ZITI_EDGE_ROUTER_IP_OVERRIDE}
+   # user specified external ip for router, use that for advertisement
+   sed -i "s/external_router_ip/${ZITI_EDGE_ROUTER_IP_OVERRIDE}/" config.yml
+   sed -i "s/advertise_ip/${ZITI_EDGE_ROUTER_IP_OVERRIDE}/" config.yml
+else
+   sed -i "s/advertise_ip/${router_ip}/" config.yml
+   sed -i "s/- \"external_router_ip\"//" config.yml
+fi
+
+#update config template with private ip of new router
+sed -i "s/router_ip/$router_ip/" config.yml
+
+#update config template with public ip of controller
+sed -i "s/edgecontroller/$controller_ip/" config.yml
+
+#update config template with edge listening port of new router
+sed -i "s/rport/$rport/" config.yml
+
+#update config template with controller fabric listening port
+sed -i "s/ctrl_port/$mgmt_port/" config.yml
+
+#update config template with name of lan interface for new router
+sed -i "s/lan_port/$lan_port/" config.yml
+
+if [ -f "ziti/ziti" ]
+then
+   echo "ziti binary found skipping download!"
+else
+   zitilatest=$(curl -s https://api.github.com/repos/openziti/ziti/releases/latest) 
```

Comment:
`ziti-cli-functions.sh` provides a `getZiti` and `getLatestZitiVersion`. any improvements should probably land in there

Reply:
OK